File: tests/fixtures/go/production_api_service.go
package service

type AppConfig struct {
	Host           string
	Port           int
	DatabaseURL    string
	RedisURL       string
	JWTSecret      string
	CORSOrigins    []string
	LogLevel       string
	MaxConnections int
	TimeoutSecs    int
	RetryCount     int
	CacheTTL       int
	RateLimit      int
	UploadDir      string
}

func DispatchCommand(cmd string) string {
	switch cmd {
	case "start":
		return "starting"
	case "stop":
		return "stopping"
	case "restart":
		return "restarting"
	case "status":
		return "checking"
	case "deploy":
		return "deploying"
	case "rollback":
		return "rolling back"
	default:
		return "unknown"
	}
}

func ProcessEvent(data []byte) int {
	a := data[0]
	b := data[1]
	c := data[2]
	d := data[3]
	e := data[4]
	f := data[5]
	g := data[6]
	h := data[7]
	if a > 100 {
		return -1
	}
	if b > 200 {
		return -2
	}
	result := int(a) + int(b) + int(c) + int(d)
	extra := int(e) + int(f) + int(g) + int(h)
	return result + extra
}

File: tests/fixtures/typescript/production_api_service.ts
interface AppConfig {
    host: string;
    port: number;
    databaseUrl: string;
    redisUrl: string;
    jwtSecret: string;
    corsOrigins: string[];
    logLevel: string;
    maxConnections: number;
    timeoutSecs: number;
    retryCount: number;
    cacheTtl: number;
    rateLimit: number;
    uploadDir: string;
}

function dispatchCommand(cmd: string): string {
    switch (cmd) {
    case "start":
        return "starting";
    case "stop":
        return "stopping";
    case "restart":
        return "restarting";
    case "status":
        return "checking";
    case "deploy":
        return "deploying";
    case "rollback":
        return "rolling back";
    default:
        return "unknown";
    }
}

function processEvent(data: number[]): number {
    let a = data[0];
    let b = data[1];
    let c = data[2];
    let d = data[3];
    let e = data[4];
    let f = data[5];
    let g = data[6];
    let h = data[7];
    if (a > 100) {
        return -1;
    }
    if (b > 200) {
        return -2;
    }
    const result = a + b + c + d;
    const extra = e + f + g + h;
    return result + extra;
}
